src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -19,51 +27,67 @@
      *                 - DAY - текущий день (число месяца);
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate date = LocalDate.now();
+        switch (datePart) {
+            case FULL:
+                return date.toString();
+            case YEAR:
+                return Integer.toString(date.getYear());
+            case MONTH:
+                return date.getMonth().toString();
+            case DAY:
+                return Integer.toString(date.getDayOfMonth());
+            //If datePart = HOURS, MINUTES or SECONDS:
+            default:
+                throw new IllegalArgumentException();
+        }
     }
 
     /**
      * Верните Optional даты соответствующей дате в массиве.
-     *
      * @param dateParams Дан массив данных состоящий из 3-х элементов, где:
      *                   - 1-й элемент массива - год;
      *                   - 2-й элемент массива - месяц;
      *                   - 3-й элемент массива - день (число);
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
+        } catch (ArrayIndexOutOfBoundsException | DateTimeException exception) {
```

Comment:
@RomanMinevich Can you please use .or here?

Reply:
In case of usage Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2])).or(Optional::empty), getDate(Integer[] dateParams) throws an exception, if any of its parameters is our of range or if dateParams[] consists of less than three elements.